From 3772abf78ce48faaac4ed807e02a7797c7b785fb Mon Sep 17 00:00:00 2001
From: Yann Gautier <yann.gautier@foss.st.com>
Date: Wed, 24 May 2023 17:01:47 +0200
Subject: [PATCH 0612/1695] plat-stm32mp2: update SRAM1 base address for
 STM32MP21

Define STM32MP25 and STM32MP21 SRAM1 based address based on
their relative configuration switch.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git aab9a7a4ac52e416f783b4c1b103d5d309087796 plat-stm32mp2: update SRAM1 base address for STM32MP21]
Signed-off-by: Yann Gautier <yann.gautier@foss.st.com>
Signed-off-by: Nicolas Le Bayon <nicolas.le.bayon@st.com>
Signed-off-by: Gatien Chevallier <gatien.chevallier@foss.st.com>
Change-Id: I18972ee47801c698dbf7de47fd6ad03919a7ea45
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/362585
Tested-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
Domain-Review: Gatien CHEVALLIER <gatien.chevallier@st.com>
Reviewed-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
---
 core/arch/arm/plat-stm32mp2/platform_config.h | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/core/arch/arm/plat-stm32mp2/platform_config.h b/core/arch/arm/plat-stm32mp2/platform_config.h
index af7ac9c78..40977cb53 100644
--- a/core/arch/arm/plat-stm32mp2/platform_config.h
+++ b/core/arch/arm/plat-stm32mp2/platform_config.h
@@ -1,6 +1,6 @@
 /* SPDX-License-Identifier: BSD-2-Clause */
 /*
- * Copyright (c) 2023, STMicroelectronics
+ * Copyright (c) 2023-2024, STMicroelectronics
  */
 
 #ifndef PLATFORM_CONFIG_H
@@ -86,7 +86,11 @@
 
 #define SYSRAM_BASE			0x0e000000
 
+#ifdef CFG_STM32MP21
+#define SRAM1_BASE			0x0e060000
+#else
 #define SRAM1_BASE			0x0e040000
+#endif
 
 /* GIC resources */
 #define GIC_SIZE			0x80000
-- 
2.47.2

